src/models/VisitSubmissionState.ts

```diff
@@ -1,11 +1,25 @@
-import { GetVisitQuery } from 'generated/sdk';
+import { ExcludeTypeName, ExcludeNull } from 'utils/utilTypes';
 
-import { VisitFragment } from './../generated/sdk';
+import {
+  GetVisitRegistrationQuery,
+  GetUserProposalBookingsWithEventsQuery,
+} from './../generated/sdk';
 import { QuestionarySubmissionState } from './QuestionarySubmissionState';
 
-export type VisitBasic = VisitFragment;
+export type RegistrationBasic = ExcludeTypeName<
+  ExcludeNull<
+    ExcludeNull<
+      ExcludeNull<
+        ExcludeNull<GetUserProposalBookingsWithEventsQuery['me']>['proposals']
+      >[0]['proposalBooking']
+    >['scheduledEvents'][0]['visit']
+  >['registrations'][0]
+>;
 
-export type VisitExtended = Exclude<GetVisitQuery['visit'], null>;
+export type RegistrationExtended = Exclude<
```

Comment:
I guess you can use the `ExcludeNull` type that you created here as well.

Reply:
Done